evaluator/evaluate.go

```diff
@@ -129,11 +130,33 @@ func updateSystemLastEvaluation(tx *gorm.DB, system *models.SystemPlatform) erro
 	return err
 }
 
-func updateSystemCaches(tx *gorm.DB, system *models.SystemPlatform) error {
+func updateCaches(tx *gorm.DB, system *models.SystemPlatform, advisoryIds []int) error {
 	tStart := time.Now()
 	defer utils.ObserveSecondsSince(tStart, evaluationPartDuration.WithLabelValues("caches-update"))
 
 	err := tx.Exec("SELECT * FROM refresh_system_caches(?,?)", system.ID, system.RhAccountID).Error
+	if err != nil {
+		return err
+	}
+
+	var acc models.RhAccount
+	err = tx.Set("gorm:query_option", "FOR UPDATE OF rh_account").
```

Comment:
When and how are we going to update rh_account table?

Reply:
This is just pessimistic locking of whole account for while we are updating the caches.